smac/facade/bogp_facade.py

```diff
@@ -38,7 +39,12 @@ def __init__(self, model_type='gp_mcmc', **kwargs):
         """
         scenario = kwargs['scenario']
         if scenario.initial_incumbent not in ['LHD', 'FACTORIAL', 'SOBOL']:
-            scenario.initial_incumbent = 'SOBOL'
+            if 'sobol' in smac.extras_installed:
```

Comment:
The behavior of SMAC should not depend on which packages are installed. We assume that all required packages are installed.

Reply:
same applies to several other changes of the code

Reply:
As a general comment, SMAC already has an (unofficial?) split between required and optional packages via the `__MANDATORY_PACKAGES__` definition and the check that these (and only these) are installed at runtime.

As for the specific change here, I assumed that the setting to `SOBOL` here was a reasonable default value but all other available options are also fine. If that's not true or we want to be stricter for backwards compatibility reasons, a runtime exception or warning are easy to add.